trin-state/src/jsonrpc.rs

```diff
@@ -102,12 +102,10 @@ impl StateRequestHandler {
                                     };
                                 match local_content {
                                     None => {
-                                        match self.network.overlay.lookup_content(content_key).await
-                                        {
-                                            Some(content) => {
-                                                let value = Value::String(hex_encode(content));
-                                                Ok(value)
-                                            }
+                                        let response =
+                                            self.network.overlay.lookup_content(content_key).await;
+                                        match response.0 {
```

Comment:
Again, I would prefer that we keep the match structure. We can always ignore irrelevant parts of the match pattern like the following:

```rust
match self.network.overlay.lookup_content(content_key).await {
  (Some(content), _) => { ... }
  ...
}
```